src/class-wp-conditional-blocks.php

```diff
@@ -8,8 +8,137 @@
 namespace Alley\WP\WP_Conditional_Blocks;
 
 /**
- * Example Plugin
+ * WP_Conditional_Blocks
  */
 class WP_Conditional_Blocks {
+	/**
+	 * Singleton instance for this class.
+	 *
+	 * @var WP_Conditional_Blocks|null
+	 */
+	private static ?self $instance = null;
 
+	/**
+	 * Array of conditions.
+	 *
+	 * @var array
+	 */
+	private array $conditions = [];
+
+	/**
+	 * Private constructor.
+	 */
+	protected function __construct() {}
+
+	/**
+	 * Creates the singleton instance.
+	 *
+	 * @return self The singleton instance.
+	 */
+	public static function instance(): self {
+		if ( null === self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Adds a condition.
+	 *
+	 * @param string   $name Name of the condition.
+	 * @param string   $slug Slug of the condition.
+	 * @param callable $condition_callback Callable function of the condition.
+	 *
+	 * @return bool True if the condition was added successfully, false otherwise.
+	 */
+	public function add_condition( string $name, string $slug, callable $condition_callback ): bool {
+		// Validate input parameters.
+		if ( empty( $name ) || empty( $slug ) || ! is_callable( $condition_callback ) ) {
+			return false;
+		}
+
+		$condition_data = [
+			'name'     => $name,
+			'slug'     => $slug,
+			'callable' => $condition_callback,
+		];
+
+		/**
+		 * Filters the condition.
+		 *
+		 * @param string $name Name of the condition.
+		 * @param string $slug Slug of the condition.
+		 * @param callable $condition_callback Callable function of the condition.
+		 */
```

Comment:
phpstan doesnt like this code comment, you can run phpstan locally to solve the errors on local too, FYI, I think it's just `vrun composer phpstan`